src/vivarium_gates_iv_iron/components/pregnancy.py

```diff
@@ -225,20 +231,34 @@ def _sample_new_pregnant(self, index: pd.Index, event_time: pd.Timestamp) -> pd.
         )
         newly_pregnant = pop.loc[not_pregnant.intersection(potentially_pregnant)].copy()
 
-        child_cols = self.new_children.columns_created
-        newly_pregnant.loc[:, child_cols] = self.new_children(newly_pregnant.index)
-
         newly_pregnant['pregnancy_status'] = models.PREGNANT_STATE
-
         p_outcome = self.outcome_probabilities(newly_pregnant.index)
-        newly_pregnant['pregnancy_outcome'] = self.randomness.choice(
+        pregnancy_outcome = self.randomness.choice(
             newly_pregnant.index,
             choices=p_outcome.columns.tolist(),
             p=p_outcome,
             additional_key='pregnancy_outcome',
         )
+        newly_pregnant['pregnancy_outcome'] = pregnancy_outcome
+
+        other_outcome = pregnancy_outcome[pregnancy_outcome == models.OTHER_OUTCOME].index
+        live_or_still_birth = pregnancy_outcome.index.difference(other_outcome)
+
+        child_cols = self.new_children.columns_created
+        newly_pregnant.loc[live_or_still_birth, child_cols] = (
+            self.new_children(live_or_still_birth)
+        )
+
+        low, high = DURATIONS.DETECTION, DURATIONS.PARTIAL_TERM
+        draw = self.randomness.get_draw(newly_pregnant.index, additional_key='pregnancy_duration')
+        pregnancy_duration = pd.to_timedelta(
+            7 * low + (high - low) * draw, unit='days'
```

Comment:
Should this read `7 * (low + (high - low) * draw)`? In other words, is it missing the parenthetical to scale the whole thing to days?